pyintegration/deephaven2/table.py

```diff
@@ -55,6 +222,801 @@ def update(self, formulas: List[str]):
+        """
+        try:
+            if by:
+                return Table(j_table=self._j_table.varBy(*by))
+            else:
+                return Table(j_table=self._j_table.varBy())
+        except Exception as e:
+            raise DHError(e, "table var_by operation failed.") from e
+
+    def median_by(self, by: List[str] = []) -> Table:
+        """ The median_by method creates a new table containing the median for each group.
+
+        Args:
+            by (List[str], optional): the group-by column names, default is empty
+
+        Returns:
+            a new table
+
+        Raises:
+            DHError
+        """
+        try:
+            if by:
+                return Table(j_table=self._j_table.medianBy(*by))
+            else:
+                return Table(j_table=self._j_table.medianBy())
+        except Exception as e:
+            raise DHError(e, "table median_by operation failed.") from e
+
+    def min_by(self, by: List[str] = []) -> Table:
+        """ The min_by method creates a new table containing the minimum value for each group.
+
+        Args:
+            by (List[str], optional): the group-by column names, default is empty
+
+        Returns:
+            a new table
+
+        Raises:
+            DHError
+        """
+        try:
+            if by:
+                return Table(j_table=self._j_table.minBy(*by))
+            else:
+                return Table(j_table=self._j_table.minBy())
+        except Exception as e:
+            raise DHError(e, "table min_by operation failed.") from e
+
+    def max_by(self, by: List[str] = []) -> Table:
+        """ The max_by method creates a new table containing the maximum value for each group.
+
+        Args:
+            by (List[str], optional): the group-by column names, default is empty
+
+        Returns:
+            a new table
+
+        Raises:
+            DHError
+        """
+        try:
+            if by:
+                return Table(j_table=self._j_table.maxBy(*by))
+            else:
+                return Table(j_table=self._j_table.maxBy())
+        except Exception as e:
+            raise DHError(e, "table max_by operation failed.") from e
+
+    def count_by(self, col: str, by: List[str] = []) -> Table:
+        """ The count_by method creates a new table containing the number of rows for each group.
+
+        Args:
+            col (str): the name of the column to store the counts
+            by (List[str], optional): the group-by column names, default is empty
+
+       Returns:
+            a new table
+
+        Raises:
+            DHError
+        """
+        try:
+            if by:
+                return Table(j_table=self._j_table.countBy(col, *by))
+            else:
+                return Table(j_table=self._j_table.countBy(col))
+        except Exception as e:
+            raise DHError(e, "table count_by operation failed.") from e
+
+    def combo_by(self, combo_agg: ComboAggregation, by: List[str]) -> Table:
```

Comment:
I think the switching parameter order is better.

Reply:
I have no strong opinion. @chipkent ?